src/AppInstallerCLITests/RestClient.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "TestCommon.h"
+#include <Rest/RestClient.h>

```

Comment:
I don't feel that these tests are sufficient.  I think we can have fairly extensive test coverage on the serialize/deserialize by writing our own http client for rest-cpp.  That will likely require creating a separate lib for the current http client code to prevent a linker conflict, but then I would hope that we can make a test one that allows us to inject our own test responses.

It may also be good experience for future work that we have, depending on the current availability of metadata about the connection.

Reply:
As discussed offline, I will be working on it and sending a different PR along with more tests.

---
In reply to: [605993882](https://github.com/microsoft/winget-cli/pull/838#discussion_r605993882) [](ancestors = 605993882)